Samples/AdventureDotNetCli/AdventureGrains/PlayerGrain.cs

```diff
@@ -199,6 +250,44 @@ private string Rest(string[] words)
             return sb.ToString().Trim().ToLower();
         }
 
+        async Task IPlayerGrain.SendMessage(string message)
+        {
+            await SendUpdateMessage(message);
+        }
+
+        async Task<string> Whisper(string words)
+        {
+            if (this.roomGrain != null)
+            {
+                await roomGrain.Whisper(words, myInfo);
+            }
+            return "You whispered '" + words + "'";
+        }
+
+        async Task<string> Shout(string words)
+        {
+            //TODO: Find a way to tell everyone in all rooms
+            if (this.roomGrain != null)
+            {
+                await roomGrain.Shout(words, myInfo);
+            }
+            
+            return "You shouted '" + words + "'";
+        }
+
+        async Task<string> Leave()
+        {
+            await DropAllItems();
+
+            if (this.roomGrain != null)
+            {
+                await roomGrain.Leave(myInfo);
+            }
+            return "You left the game";
+
+            this.DeactivateOnIdle(); // Force a deactivation of the grain
```

Comment:
This is unreachable code, please remove or call before the return statement if you intend to deactivate.